### PR TITLE
[Mold] Use the official libexecdir now that the CCI hooks allow it

### PR DESCRIPTION
**Pull request by ericriff**
Follow up to https://github.com/conan-io/conan-center-index/pull/13666

`mold` wants to install a symlink in `packagefolder/libexec`, but this folder is forbidden by the CCI hooks, particularly by `KB-H013`. The PR linked above patched the mold sources to make this symlink get installed somewhere else, which is not ideal since this path is explicitly mentioned on the docs.
@prince-chrismc suggested patching the hooks instead so this rule doesn't get enforced on `mold`. Now that https://github.com/conan-io/hooks/pull/464 has been merged, I'm removing the patch on the mold recipe and making sure the symlink gets installed on the proper directory.

Thanks @prince-chrismc once again for the suggestion.

### CODE REVIEW

**Comment by ericriff** [comment]
BTW this problem with the test package and the generators still persist for me. Can one of the experts take a look? :pray: 
https://github.com/conan-io/conan-center-index/pull/13666#issuecomment-1344727527

**Comment by ericriff** [comment]
Well, I have no idea what's going on. It was complaining about `zlib/1.2.12` not being found on conan v2, so I bumped it to `13`, but nothing changed.

**Comment by prince-chrismc** [comment]
The v2 pipeline is running a super duper old version that doesn't work so there's no packages.

Just ignore that for now

File: recipes/mold/all/conanfile.py [hunk 1]
```diff
@@ -64,6 +61,7 @@ def generate(self):
         tc.variables["MOLD_USE_MIMALLOC"] = self.options.with_mimalloc
         tc.variables["MOLD_USE_SYSTEM_MIMALLOC"] = True
         tc.variables["MOLD_USE_SYSTEM_TBB"] = True
+        tc.variables["CMAKE_INSTALL_LIBEXECDIR"] = "libexec"
```
[the remark this answers is not in the document]
Comment (prince-chrismc) [reply recipes/mold/all/conanfile.py:64]: I wonder if the cmake layout of helper are overriding this 🤔

**Comment by ericriff** [comment]
Thanks @prince-chrismc . I need to learn to read. It clearly says `(informative, not required for merge)` haha
Do you have any comments about https://github.com/conan-io/conan-center-index/pull/15068#issuecomment-1369036645?

File: recipes/mold/all/conanfile.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (prince-chrismc) [reply recipes/mold/all/conanfile.py:64]: https://github.com/search?q=repo%3Aconan-io%2Fconan%20libexec&type=code

It is

[the remark this answers is not in the document]
Comment (prince-chrismc) [reply recipes/mold/all/conanfile.py:64]: I asked the Conan team and @jcar87 said this should not be any reason of concern, so we are good :)

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
I am still learning two profiles myself but AFAIK, "only host" means everything goes into the same scope... but adding the build profile we seperate things into two scopes

https://github.com/conan-io/conan-center-index/blob/7260c49e7020931ddb07f2e04941f6bb5711a240/recipes/mold/all/test_package/conanfile.py

https://github.com/conan-io/conan-center-index/blob/7260c49e7020931ddb07f2e04941f6bb5711a240/recipes/mold/all/test_package/conanfile.py#L15 should have a "build" scope AFAIK

https://github.com/conan-io/conan-center-index/blob/c2502235b71dbb84a10b5f5df9b8084847e9707e/recipes/cppfront/all/test_package/conanfile.py#L26

**Pull request merged**